Tests/GRPCNIOTransportCoreTests/Client/GRPCClientStreamHandlerTests.swift

```diff
@@ -161,6 +179,16 @@ final class GRPCClientStreamHandlerTests: XCTestCase {
         Metadata(headers: serverInitialMetadata)
       )
     )
+
+    // We should not throw if the server sends another message:
+    // we should drop it, since the server is now closed.
+    var buffer = ByteBuffer()
+    buffer.writeInteger(UInt8(0))  // not compressed
+    buffer.writeInteger(UInt32(42))  // message length
+    buffer.writeRepeatingByte(0, count: 42)  // message
+    let serverDataPayload = HTTP2Frame.FramePayload.Data(data: .byteBuffer(buffer), endStream: true)
```

Comment:
What about the server sending trailers with end stream set? We should cover that case as well.